craft_providers/util/env_cmd.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helper(s) for env command."""
+from typing import Dict, List, Optional
+
+
+def formulate_command(
+    env: Optional[Dict[str, Optional[str]]] = None, *, ignore_environment: bool = False
+) -> List[str]:
+    """Create an env command with the specified environment.
+
+    For each key-value, the env command will include the key=value argument to
+    the env command.
+
+    If a variable is None, then the env -u parameter will be used to unset it.
+
+    An empty environment will simply yield the env command.
+
+    :param env: Environment flags to set/unset.
+    :param ignore_environment: Start with an empty environment.
+
+    :returns: List of env command strings.
+    """
+    env_cmd = ["env"]
+
+    if ignore_environment:
+        env_cmd.append("-i")
+
+    if env is None:
```

Comment:
Does it make sense to ignore the environment, and also set env vars?

I assume it doesn't. What do you think  about `assert env is None` in the block above, and directly return env_cmd without ending the `if`? That way we cover for a potential misuse of the function by mistake.

Reply:
It does if you want env to mask the default environment and initialize env with only with what you specify.  I've personally used it in a number of cases in the past.

Whether or not we'll use it here, I am unsure.  I just added it for completeness should we want to.

Reply:
Ah, it ignores the *previous* environment. Ok.